### ISSUE TITLE
Support poe.ninja currency ratios and trends

### ISSUE THREAD

**Issue by domialex** [domialex/Sidekick#5]

**Comment by kai-oswald** [comment]
Currently working on this! Might have a working version later today.

**Comment by domialex** [comment]
> Currently working on this! Might have a working version later today.

Do you still have time to update your PR? Considering we now use a different architecture it should have a smaller impact.

**Comment by kai-oswald** [comment]
I'll have something soon™ (later today).

**Comment by kai-oswald** [comment]
I'm having issues with the DI, @domialex maybe you can take a look.
I've merged the most recent version of this develop branch to my fork.

The main problem I have is with `IInitializableService` and injecting services into those.

I set it up like this:
```csharp
services.AddSingleton<PoeNinjaClient>();
services.AddInitializableService<IPoeNinjaCache, PoeNinjaCache>();
```
Then the `PoeNinjaCache`:
```csharp
public class PoeNinjaCache : IPoeNinjaCache, IOnBeforeInit
{
        private readonly PoeNinjaClient _client;
        public PoeNinjaCache(PoeNinjaClient client)
        {
            _client = client;
        }
}
```

This is where the dependency to `PoeNinjaClient` cannot be resolved.

The fork (maybe try merging it locally to your develop branch):
https://github.com/kai-oswald/Sidekick/tree/develop

**Comment by leMicin** [comment]
Also you may be missing the PoeNinjaClient in your services. PoeNinjaClient should be registered to the container also, and have an interface. If the client is a static class, it needs refactoring also.

**Comment by leMicin** [comment]
Also: Here is some documentation on the design pattern :
https://docs.microsoft.com/en-us/aspnet/core/fundamentals/dependency-injection?view=aspnetcore-3.1

**Comment by kai-oswald** [comment]
Was an issue where I didn't adjust the league, but just fyi you can also register services without the need of an interface.

**Comment by leMicin** [comment]
I understand that you can register a service without an interface, though it doesn't give us the full benefits of using Dependency Injection. Sepcifically this point from the documentation : "This implementation is difficult to unit test. The app should use a mock or stub MyDependency class, which isn't possible with this approach."

**Comment by domialex** [comment]
Implemented with #137.
